src/wrapper.js

```diff
@@ -20,25 +20,53 @@
 
 		// For CommonJS and CommonJS-like environments where a proper `window`
 		// is present, execute the factory and get jQuery.
-		// For environments that do not have a `window` with a `document`
+		// For environments that do not have a global `document`
 		// (such as Node.js), expose a factory as module.exports.
-		// This accentuates the need for the creation of a real `window`.
-		// e.g. var jQuery = require("jquery")(window);
-		// See ticket #14549 for more info.
-		module.exports = global.document ?
-			factory( global, true ) :
-			function( w ) {
-				if ( !w.document ) {
-					throw new Error( "jQuery requires a window with a document" );
+
+		module.exports = ( global.window && global.window.document ) ?
+			factory( global.window, true, global ) :
+			function( global, window ) {
```

Comment:
This line is what I was talking about - you require jQuery Node.js users to pass the global object as the first parameter. First of all, that would be a breaking change since we now expect `window` to be the first so - if anything - the order of parameters should be swapped.

But I still don't see the point of passing the `global` to the factory. jQuery itself doesn't need it so if the only reason is to pass the object to which `jQuery` & `$` are attached, that doesn't justify complicating the API. If someone wants those variables to be accessible via the global, they can attach them themselves:

```js
var $ = require( "jquery" );
global.$ = global.jQuery = $;
```